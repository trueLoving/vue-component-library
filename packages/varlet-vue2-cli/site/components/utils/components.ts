import Vue, { ComponentOptions } from 'vue'
import { CombinedVueInstance } from 'vue/types/vue'

export function pickProps<T, U extends keyof T>(props: T, propsKey: U): T[U]
export function pickProps<T, U extends keyof T>(props: T, propsKey: U[]): Pick<T, U>
export function pickProps(props: any, propsKey: any): any {
  return Array.isArray(propsKey)
    ? propsKey.reduce((pickedProps: any, key) => {
        pickedProps[key] = props[key]
        return pickedProps
      }, {})
    : props[propsKey]
}

export interface MountComponentApi {
  instance: CombinedVueInstance<any, any, any, any, any>
  unmount(): void
}

export function mountComponent(component: ComponentOptions<Vue>, container = 'body', options = {}): MountComponentApi {
  const instance = new (Vue.extend(component))(options)
  const el = instance.$mount().$el
  const wrapper = document.querySelector(container)!
  wrapper.appendChild(el)

  return {
    instance,
    unmount() {
      instance.$destroy()
      wrapper.removeChild(el)
    },
  }
}
