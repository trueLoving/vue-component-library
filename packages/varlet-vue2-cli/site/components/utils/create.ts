import Vue, { ComponentOptions, RenderContext, VueConstructor } from 'vue'
import { SlotsMixin } from './mixins/slots'
import { ListenersMixin } from './mixins/listeners'

export type ComponentOptionsPlugin = ComponentOptions<Vue> & { install?: (app: VueConstructor) => void }

export function nextTick() {
  return new Promise(Vue.nextTick)
}

export function mergeSlots(context: RenderContext) {
  const scopedSlots = context.scopedSlots ?? context.data.scopedSlots ?? {}
  const slots = context.slots()

  Object.keys(slots).forEach((key) => {
    if (!scopedSlots[key]) {
      scopedSlots[key] = () => slots[key]
    }
  })

  return scopedSlots
}

export function defineComponent(sfc: ComponentOptionsPlugin & { name: string }) {
  if (sfc.mixins) {
    sfc.mixins.push(SlotsMixin)
    sfc.mixins.push(ListenersMixin)
  } else {
    sfc.mixins = [SlotsMixin, ListenersMixin]
  }

  return sfc
}
