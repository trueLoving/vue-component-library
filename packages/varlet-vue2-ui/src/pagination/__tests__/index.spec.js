import example from '../example'
import Pagination from '..'
import VarPagination from '../Pagination'
import Vue from 'vue'
import { mount } from '@vue/test-utils'
import { delay, trigger } from '../../utils/jest'

test('test pagination example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test pagination plugin', () => {
  Vue.use(Pagination)
  expect(Vue.component(Pagination.name)).toBeTruthy()
})

describe('simple mode', () => {
  test('mount', () => {
    const wrapper = mount(VarPagination, {
      propsData: {
        total: '123',
        showTotal(total, range) {
          return `${range[0]}-${range[1]} of ${total} items`
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.destroy()
  })

  test('disabled prop', () => {
    const wrapper = mount(VarPagination, {
      propsData: {
        disabled: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.destroy()
  })
})

describe('basic mode', () => {
  test('mount', () => {
    const wrapper = mount(VarPagination, {
      propsData: {
        total: '123',
        simple: false,
        showQuickJumper: true,
        showTotal(total, range) {
          return `${range[0]}-${range[1]} of ${total} items`
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.destroy()
  })

  test('disabled prop', () => {
    const wrapper = mount(VarPagination, {
      propsData: {
        disabled: true,
        simple: false,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.destroy()
  })

  test('click ellipsis', async () => {
    const change = jest.fn()
    const template = `
      <var-pagination
        current="10"
        :size="20"
        total="325"
        :max-pager-count="4"
        :simple="false"
        @change="change"
      />
    `
    const wrapper = mount({
      components: {
        [VarPagination.name]: VarPagination,
      },

      methods: {
        change,
      },
      template,
    })

    await delay(0)

    const prev = wrapper.find('.var-pagination__item[item-mode="head"]')
    const next = wrapper.find('.var-pagination__item[item-mode="tail"]')

    expect(wrapper.find('.var-pagination__item-active').text()).toBe('10')

    await next.trigger('click')
    expect(wrapper.find('.var-pagination__item-active').text()).toBe('14')

    await prev.trigger('click')
    expect(wrapper.find('.var-pagination__item-active').text()).toBe('10')

    expect(change).toBeCalledTimes(2)
    wrapper.destroy()
  })
})

test('Size related props', async () => {
  const template = `<var-pagination total="115" size="20" :show-size-changer="sizeChange" :size-option="sizeOption"/>`

  const wrapper = mount({
    components: {
      [VarPagination.name]: VarPagination,
    },
    data() {
      return {
        sizeChange: true,
        sizeOption: [10, 20, 30, 40],
      }
    },
    template,
  })

  await delay(0)

  expect(wrapper.html()).toMatchSnapshot()

  const lists = document.querySelectorAll('.var-pagination__list')

  expect(lists.length).toBe(4)

  expect(lists[3].innerHTML.includes('40')).toBeTruthy()

  await wrapper.setData({ sizeChange: false })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('change current and size', async () => {
  const wrapper = mount(VarPagination, {
    propsData: {
      current: 4,
      size: 10,
      total: 300,
    },
  })

  const input = wrapper.find('.var-input__input')
  const sizeEl = wrapper.find('.var-pagination__size-open')

  expect(input.element.value).toBe('4')
  expect(sizeEl.text().includes('10')).toBe(true)

  await wrapper.setProps({
    current: 5,
    size: 20,
  })

  expect(input.element.value).toBe('5')
  expect(sizeEl.text().includes('20')).toBe(true)

  wrapper.destroy()
})

test('change event', async () => {
  const change = jest.fn()
  const template = `<var-pagination current="2" :size="20" total="325" @change="change" />`

  const wrapper = mount({
    components: {
      [VarPagination.name]: VarPagination,
    },

    methods: {
      change,
    },
    template,
  })

  await delay(0)

  const items = wrapper.findAll('.var-pagination__item')
  const next = wrapper.find('.var-pagination__next')
  const input = wrapper.find('.var-input__input')

  await input.trigger('blur')
  expect(input.element.value).toBe('2')

  await items.at(0).trigger('click')
  expect(input.element.value).toBe('1')

  await items.at(1).trigger('click')
  expect(input.element.value).toBe('2')
  expect(change).toBeCalledTimes(2)

  await wrapper.find('.var-pagination__size-open').trigger('click')
  const li = document.querySelector('.var-cell__title')
  await trigger(li, 'click')

  expect(change).toBeCalledTimes(3)
  expect(wrapper.find('.var-pagination__simple').text().includes('33')).toBe(true)
  wrapper.destroy()
})
