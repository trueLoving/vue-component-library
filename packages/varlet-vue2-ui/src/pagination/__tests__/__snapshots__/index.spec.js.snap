// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Size related props 1`] = `
"<ul class=\\"var-pagination\\">
  <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__simple\\">
    <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
      <div class=\\"var-input__controller\\">
        <div class=\\"var-input__icon\\"></div>
        <div class=\\"var-input__wrap\\">
          <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-127\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-127\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

          </label></div>
        <div class=\\"var-input__icon\\">
          <!---->
        </div>
      </div>
      <div class=\\"var-input__line\\">
        <div class=\\"var-input__dot\\"></div>
      </div>
      <transition-stub name=\\"var-form-details\\">
        <!---->
      </transition-stub>
    </div> <span>/ 6</span>
  </li>
  <li class=\\"var-pagination__item var-pagination__next var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__size\\">
    <div class=\\"var-menu\\">
      <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>20条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
      <div class=\\"var-teleport\\"></div>
    </div>
  </li>
  <!---->
  <!---->
</ul>"
`;

exports[`Size related props 2`] = `
"<ul class=\\"var-pagination\\">
  <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__simple\\">
    <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
      <div class=\\"var-input__controller\\">
        <div class=\\"var-input__icon\\"></div>
        <div class=\\"var-input__wrap\\">
          <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-127\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-127\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

          </label></div>
        <div class=\\"var-input__icon\\">
          <!---->
        </div>
      </div>
      <div class=\\"var-input__line\\">
        <div class=\\"var-input__dot\\"></div>
      </div>
      <transition-stub name=\\"var-form-details\\">
        <!---->
      </transition-stub>
    </div> <span>/ 6</span>
  </li>
  <li class=\\"var-pagination__item var-pagination__next var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
  <!---->
  <!---->
  <!---->
</ul>"
`;

exports[`basic mode disabled prop 1`] = `
"<ul class=\\"var-pagination\\">
  <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-elevation--2\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__item var-pagination__next var-pagination__item-disabled var-elevation--2\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__size var-pagination__item-disabled\\">
    <div class=\\"var-menu\\">
      <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
      <transition-stub name=\\"var-menu\\">
        <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
          <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                10条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                20条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                50条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                100条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </transition-stub>
    </div>
  </li>
  <!---->
  <!---->
</ul>"
`;

exports[`basic mode mount 1`] = `
"<ul class=\\"var-pagination\\">
  <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-elevation--2\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li item-mode=\\"basic\\" class=\\"var-pagination__item var-elevation--2 var-pagination__item-active\\">
    1
  </li>
  <li item-mode=\\"head\\" class=\\"var-pagination__item var-elevation--2 var-pagination__item-hide\\">
    ...
  </li>
  <li item-mode=\\"basic\\" class=\\"var-pagination__item var-elevation--2\\">
    2
  </li>
  <li item-mode=\\"basic\\" class=\\"var-pagination__item var-elevation--2\\">
    3
  </li>
  <li item-mode=\\"basic\\" class=\\"var-pagination__item var-elevation--2\\">
    4
  </li>
  <li item-mode=\\"tail\\" class=\\"var-pagination__item var-elevation--2\\">
    ...
  </li>
  <li item-mode=\\"basic\\" class=\\"var-pagination__item var-elevation--2\\">
    13
  </li>
  <li class=\\"var-pagination__item var-pagination__next var-elevation--2\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__size\\">
    <div class=\\"var-menu\\">
      <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
      <transition-stub name=\\"var-menu\\">
        <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
          <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                10条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                20条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                50条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                100条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </transition-stub>
    </div>
  </li>
  <li class=\\"var-pagination__quickly\\">
    前往
    <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
      <div class=\\"var-input__controller\\">
        <div class=\\"var-input__icon\\"></div>
        <div class=\\"var-input__wrap\\">
          <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-90\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-90\\" class=\\"var--ellipsis var-input__placeholder\\">

          </label></div>
        <div class=\\"var-input__icon\\">
          <!---->
        </div>
      </div>
      <div class=\\"var-input__line\\">
        <div class=\\"var-input__dot\\"></div>
      </div>
      <transition-stub name=\\"var-form-details\\">
        <!---->
      </transition-stub>
    </div>
  </li>
  <li class=\\"var-pagination__total\\">
    1-10 of 123 items
  </li>
</ul>"
`;

exports[`simple mode disabled prop 1`] = `
"<ul class=\\"var-pagination\\">
  <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__simple var-pagination__item-disabled\\">
    <div class=\\"var-input var--box var-input--disabled\\" var-pagination-cover=\\"\\">
      <div class=\\"var-input__controller var-input--disabled\\">
        <div class=\\"var-input__icon\\"></div>
        <div class=\\"var-input__wrap\\">
          <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-68\\" disabled=\\"disabled\\" type=\\"text\\" class=\\"var-input__input var-input--disabled\\"> <label for=\\"var-input-68\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

          </label></div>
        <div class=\\"var-input__icon\\">
          <!---->
        </div>
      </div>
      <div class=\\"var-input__line var-input--line-disabled\\">
        <div class=\\"var-input__dot var-input--line-disabled\\"></div>
      </div>
      <transition-stub name=\\"var-form-details\\">
        <!---->
      </transition-stub>
    </div> <span>/ 0</span>
  </li>
  <li class=\\"var-pagination__item var-pagination__next var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__size var-pagination__item-disabled\\">
    <div class=\\"var-menu\\">
      <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
      <transition-stub name=\\"var-menu\\">
        <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
          <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                10条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                20条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                50条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                100条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </transition-stub>
    </div>
  </li>
  <!---->
  <!---->
</ul>"
`;

exports[`simple mode mount 1`] = `
"<ul class=\\"var-pagination\\">
  <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__simple\\">
    <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
      <div class=\\"var-input__controller\\">
        <div class=\\"var-input__icon\\"></div>
        <div class=\\"var-input__wrap\\">
          <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-54\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-54\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

          </label></div>
        <div class=\\"var-input__icon\\">
          <!---->
        </div>
      </div>
      <div class=\\"var-input__line\\">
        <div class=\\"var-input__dot\\"></div>
      </div>
      <transition-stub name=\\"var-form-details\\">
        <!---->
      </transition-stub>
    </div> <span>/ 13</span>
  </li>
  <li class=\\"var-pagination__item var-pagination__next var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
  <li class=\\"var-pagination__size\\">
    <div class=\\"var-menu\\">
      <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
      <transition-stub name=\\"var-menu\\">
        <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
          <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                10条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                20条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                50条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"var-cell var-pagination__list\\">
            <!---->
            <div class=\\"var-cell__content\\">
              <div class=\\"var-cell__title\\">
                100条 / 页
              </div>
              <!---->
            </div>
            <!---->
          </div>
        </div>
      </transition-stub>
    </div>
  </li>
  <!---->
  <li class=\\"var-pagination__total\\">
    1-10 of 123 items
  </li>
</ul>"
`;

exports[`test pagination example 1`] = `
"<div class=\\"var-pagination-example\\">
  <div class=\\"app-type\\">基本使用</div>
  <div class=\\"transformer\\">
    <ul class=\\"var-pagination\\">
      <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__simple\\">
        <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
          <div class=\\"var-input__controller\\">
            <div class=\\"var-input__icon\\"></div>
            <div class=\\"var-input__wrap\\">
              <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-5\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-5\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

              </label></div>
            <div class=\\"var-input__icon\\">
              <!---->
            </div>
          </div>
          <div class=\\"var-input__line\\">
            <div class=\\"var-input__dot\\"></div>
          </div>
          <transition-stub name=\\"var-form-details\\">
            <!---->
          </transition-stub>
        </div> <span>/ 12</span>
      </li>
      <li class=\\"var-pagination__item var-pagination__next var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__size\\">
        <div class=\\"var-menu\\">
          <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
          <transition-stub name=\\"var-menu\\">
            <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
              <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    10条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    20条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    50条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    100条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </transition-stub>
        </div>
      </li>
      <!---->
      <!---->
    </ul>
  </div>
  <div class=\\"app-type\\">隐藏 size 切换器</div>
  <div class=\\"transformer\\">
    <ul class=\\"var-pagination\\">
      <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__simple\\">
        <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
          <div class=\\"var-input__controller\\">
            <div class=\\"var-input__icon\\"></div>
            <div class=\\"var-input__wrap\\">
              <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-19\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-19\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

              </label></div>
            <div class=\\"var-input__icon\\">
              <!---->
            </div>
          </div>
          <div class=\\"var-input__line\\">
            <div class=\\"var-input__dot\\"></div>
          </div>
          <transition-stub name=\\"var-form-details\\">
            <!---->
          </transition-stub>
        </div> <span>/ 12</span>
      </li>
      <li class=\\"var-pagination__item var-pagination__next var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
      <!---->
      <!---->
      <!---->
    </ul>
  </div>
  <div class=\\"app-type\\">显示总数</div>
  <div class=\\"transformer\\">
    <ul class=\\"var-pagination\\">
      <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__simple\\">
        <div class=\\"var-input var--box\\" var-pagination-cover=\\"\\">
          <div class=\\"var-input__controller\\">
            <div class=\\"var-input__icon\\"></div>
            <div class=\\"var-input__wrap\\">
              <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-26\\" type=\\"text\\" class=\\"var-input__input\\"> <label for=\\"var-input-26\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

              </label></div>
            <div class=\\"var-input__icon\\">
              <!---->
            </div>
          </div>
          <div class=\\"var-input__line\\">
            <div class=\\"var-input__dot\\"></div>
          </div>
          <transition-stub name=\\"var-form-details\\">
            <!---->
          </transition-stub>
        </div> <span>/ 12</span>
      </li>
      <li class=\\"var-pagination__item var-pagination__next var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__size\\">
        <div class=\\"var-menu\\">
          <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
          <transition-stub name=\\"var-menu\\">
            <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
              <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    10条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    20条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    50条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    100条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </transition-stub>
        </div>
      </li>
      <!---->
      <li class=\\"var-pagination__total\\">
        共 120 条
      </li>
    </ul>
  </div>
  <div class=\\"app-type\\">禁用</div>
  <div class=\\"transformer\\">
    <ul class=\\"var-pagination\\">
      <li class=\\"var-pagination__item var-pagination__prev var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__simple var-pagination__item-disabled\\">
        <div class=\\"var-input var--box var-input--disabled\\" var-pagination-cover=\\"\\">
          <div class=\\"var-input__controller var-input--disabled\\">
            <div class=\\"var-input__icon\\"></div>
            <div class=\\"var-input__wrap\\">
              <!----> <input autocomplete=\\"new-password\\" id=\\"var-input-40\\" disabled=\\"disabled\\" type=\\"text\\" class=\\"var-input__input var-input--disabled\\"> <label for=\\"var-input-40\\" class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\">

              </label></div>
            <div class=\\"var-input__icon\\">
              <!---->
            </div>
          </div>
          <div class=\\"var-input__line var-input--line-disabled\\">
            <div class=\\"var-input__dot var-input--line-disabled\\"></div>
          </div>
          <transition-stub name=\\"var-form-details\\">
            <!---->
          </transition-stub>
        </div> <span>/ 12</span>
      </li>
      <li class=\\"var-pagination__item var-pagination__next var-pagination__item-disabled var-pagination__item-hover\\"><i class=\\"var-icon var-icon--set var-icon-chevron-right\\" style=\\"transition: transform 0ms;\\"></i></li>
      <li class=\\"var-pagination__size var-pagination__item-disabled\\">
        <div class=\\"var-menu\\">
          <div class=\\"var-pagination__size-open\\" style=\\"display: flex;\\"><span>10条 / 页</span> <i class=\\"var-icon var-pagination__size-open-icon var-icon--set var-icon-menu-down\\" style=\\"transition: transform 0ms;\\" var-pagination-cover=\\"\\"></i></div>
          <transition-stub name=\\"var-menu\\">
            <div class=\\"var-menu__menu var-elevation--3\\" style=\\"top: calc(0px + 0px); left: calc(0px + -4px); z-index: 2000; display: none;\\">
              <div class=\\"var-cell var-pagination__list var-pagination__list-active\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    10条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    20条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    50条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
              <div class=\\"var-cell var-pagination__list\\">
                <!---->
                <div class=\\"var-cell__content\\">
                  <div class=\\"var-cell__title\\">
                    100条 / 页
                  </div>
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </transition-stub>
        </div>
      </li>
      <!---->
      <!---->
    </ul>
  </div>
</div>"
`;
